.github/CONTRIBUTING.md

```diff
@@ -59,7 +59,180 @@ Modifying the schema requires not only direct edits to the data models (found in
 
 ### Testing
-"Software engineering is the integral of coding over time"; to build a reliable system over several years, we rely on tests to verify that we've implemented changes correctly without breaking anything unexpected in the process, that we throw errors when appropriate, and that we've handled forgettable edge cases. This is currently a bit of a work in progress, but it's something to keep in mind as you're developing&mdash;the use cases to be tested&mdash;and once we've established some examples and work towards testing the entire system, we'll expect you to write tests for your own PRs.
+"Software engineering is the integral of coding over time"; to build a reliable system over several years, we rely on tests to verify that we've implemented changes correctly without breaking anything unexpected in the process, that we throw errors when appropriate, and that we've handled forgettable edge cases. 
+
+The tests we've defined can be found in the `tests` folder, and currently contain tests for the API layer. These tests are written with [Jest](https://jestjs.io/), with mocking done with (ts-mockito)[https://github.com/NagRock/ts-mockito] (a TypeScript version of the popular [Mockito](https://site.mockito.org/) library in Java). We don't test specific functions in the service or repository layer, since given the heirarchical structure of our layers, correctness at the API layer implies correctness in the service and repository layers.
+
+#### Test Design
+
+Our tests are a mix of unit and integration tests &mdash; we will often test that specific API routes work as intended and can handle different scenarios as a unit, but we also test that the API route works in tandem with other routes if a given user flow uses those routes. For example, in our merch store, we have a feature where users can cancel orders that they placed. An example unit test could be: 'cancelling an order refunds the user the amount of credits they spent on that order'. However, there are other interactions that can depend on this route, such as if a user has their order fulfilled. So a more 'integrated' test could be: 'cancelling an already fulfilled order is not allowed', which uses both the 'cancel order' and 'fulfill order' functionalities in a single test.
+
+#### Test Structure
+Every test consists of 3 main parts: setup, execution, and assertion. The setup step involves creating any data that is needed for that specific functionality to be tested. Below is an example merch store test that ensures members earn both points and credits for attending an event. The setup involves retrieving a `Connection` object via. the `DatabaseConnection` class (so we can write data to the database later on), and creating a fake member to check into a fake event. Every test will need to start by getting a `Connection` object, because the actual operations that take place on persistent data depend on it (see section on `ControllerFactory`).
```

Comment:
We highlight the 3 parts of a test but the following sections aren't very clear on what section corresponds to where - maybe we could add subtitles fot each part

(i.e. under setup there is factories, portal state)

Reply:
Also, mocking is more in the execution step, shouldn't it be in the setup with the factories?

Reply:
> We highlight the 3 parts of a test but the following sections aren't very clear on what section corresponds to where - maybe we could add subtitles fot each part

Good point, updated the sections with subtitles.

> Also, mocking is more in the execution step, shouldn't it be in the setup with the factories?

While I partially agree here, I think mocking is more directly tied to the execution step, since we need to know what services are being executed, mock those services, and pass the mocked instance into the `ControllerFactory` call, which is used primarily in the execution step.